packages/babel-parser/src/parser/node.js

```diff
@@ -14,7 +14,9 @@ class Node implements NodeBase {
     this.end = 0;
     this.loc = new SourceLocation(loc);
     if (parser?.options.ranges) this.range = [pos, 0];
-    if (parser?.filename) this.loc.filename = parser.filename;
+    if (parser?.filename && parser?.filename !== "unknown") {
```

Comment:
Comparing string during every AST node construction is slow. I think as a parser we should do option normalizing in the parser constructor and trust the input in node construction.

Reply:
Yep, already done that, just forgot to commit it